### PR TITLE
openjdk: add livecheck block

### PR DESCRIPTION
**Pull request by J-M0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/openjdk.rb [hunk 1]
```diff
@@ -5,6 +5,11 @@ class Openjdk < Formula
   sha256 "9c5be662f5b166b5c82c27de29b71f867cff3ff4570f4c8fa646490c4529135a"
   license :cannot_represent
 
+  livecheck do
+    url "https://hg.openjdk.java.net/jdk-updates/jdk15u/tags"
```
Comment (carlocab) [inline Formula/openjdk.rb:9]:
```suggestion
    url "https://hg.openjdk.java.net/jdk-updates/jdk#{version.major}u/tags"
```
Can you see if this will work?

Comment (J-M0) [reply Formula/openjdk.rb:9]: It works on my machine

Comment (J-M0) [reply Formula/openjdk.rb:9]: I'll push the changes

File: Formula/openjdk.rb [hunk 2]
```diff
@@ -5,6 +5,11 @@ class Openjdk < Formula
   sha256 "9c5be662f5b166b5c82c27de29b71f867cff3ff4570f4c8fa646490c4529135a"
   license :cannot_represent
 
+  livecheck do
+    url "https://hg.openjdk.java.net/jdk-updates/jdk#{version.major}u/tags"
+    regex(/jdk-(\d+\.?\d*\.?\d*)-ga/i)
```
Comment (samford) [inline Formula/openjdk.rb:10]:
```suggestion
    regex(/>\s*?jdk[._-]v?(\d+(?:\.\d+)*)-ga\s*?</i)
```

This brings the regex more in line with existing standards and anchors the beginning/end, so we only match the text in links (not any text on the page).

**Review by samford — CHANGES_REQUESTED** [review]
We recently added the ability to reference `version` in `livecheck` blocks (as part of adding cask support) but I haven't had the time to revisit related formulae yet. `openjdk` has long been one of the formulae that are hard to create a check for (due to the upstream situation and/or technical limitations in livecheck).

The check here is in line with my previous attempts but I never committed my work because I wasn't satisfied using a URL with a static version in it. This addresses that shortcoming, at least, so it's a bit more appealing now. Unfortunately, this check still won't be able to notify us of a new major version (only minor/patch versions) but this is the best we can do, so I'm fine moving forward with it.

Once the requested change is integrated this should be good to merge.

File: Formula/openjdk.rb (hunk 2 above)
Comment (J-M0) [reply Formula/openjdk.rb]: Done

**Review by samford — APPROVED** [review]
Thanks, @J-M0!

**Pull request merged**